gettingstarted/index.md

```diff
@@ -2,7 +2,462 @@
+  }
+
+  Ok()
+}
+```
+
+API route: `DELETE /api/users.v1/:id`
+
+A Delete RestAction expects a return type of `RestResponse[Unit]`.
+
+Naptime expects that the Delete method argument is named `id`. The resource ID
+extracted from the last part of the URL will be passed into the method as the
+`id` parameter.
+
+#### Multi Get
+
+It's useful for client to be able to batch together requests for multiple items
+for efficiency reasons. Naptime supports this via the multiGet API.
+
+```scala
+def multiGet(ids: Set[Int]) = Rest.multiGet { context =>
+
+  val keyedUsers = userStore.get(ids)
+
+  Ok(keyedUsers.toList)
+}
+```
+
+API route: `GET /api/users.v1?ids=1,3,5`
+
+A MultiGet RestAction expects a return type of `RestResponse[Seq[Keyed[Int, User]]]`.
+
+#### Get All / Pagination
+
+Naptime supports listing all entities in a collection. When the number of items
+is potentially large, pagination should definitely be used.
+
+```scala
+def getAll() = Rest.getAll { context =>
+
+  val paging = context.paging
+  val (nextKey, keyedUsers) = userStore.getAll(
+    paging.startAsInt.getOrElse(0),
+    paging.limit)
+
+  Ok(keyedUsers, pagination = Some(ResponsePagination(nextKey.map(_.toString))))
+}
+```
+
+API route: `GET /api/users.v1`
+
+A GetAll RestAction expects a return type of `RestResponse[Seq[Keyed[Int, User]]]`.
+
+Let's take a closer look at pagination. The request pagination parameters are
+availabe in `RestContext.paging`, and which contains the fields `start` and
+`limit`. The `ResponsePagination` returns an optional `start` which indicates
+whether more data can be obtained by calling with the same API with updated
+pagination.
+
+Pagination is intentionally designed with `start` being an opaque `String` type,
+as opposed to a numeric page number, so as to support datastores / collections
+that do not easily support numeric paging. However, there is a convenience
+function `startAsInt` that will attempt to parse `start` as an Int, and throw a
+BadRequest exception on failure.
+
+#### Finders - Finding things
+
+Sometimes we need a way of finding entities other than by key. Finders to the
+rescue!
+
+Finders are purely to find things – they should not have side effects.
+
+Finders take the form of `GET /api/users.v1?q=finderName&param1=value1&param2=value2`,
+and you would write a different finder for each type of query you're aiming to
+support.
+
+Let's write a finder to lookup users by email:
+
+```scala
+def email(email: String) = Rest.finder { context =>
+
+  val keyedUsers = userStore.findByEmail(email)
+
+  Ok(keyedUsers.toList)
+}
+```
+
+API route: `GET /api/users.v1?q=email&email=a@123.com`
+
+A Finder RestAction expects a return type of `RestResponse[Seq[Keyed[Int, User]]]`.
+
+Finders are named after the function defining them (in this case, our finder is
+called `email`). Query parameters are dynamically bound to the finder arguments,
+and are parsed to the right type. Custom key types (using `KeyFormat`) are
```

Comment:
Ok. When `KeyFormat` docs are included, it might be nice to define `UserId` for this example to show how to use non-primitive key types.